src/m365/aad/commands/m365group/m365group-remove.ts

```diff
@@ -115,6 +109,124 @@ class AadM365GroupRemoveCommand extends GraphCommand {
       }
     }
   }
+
+  private async getM365GroupSiteURL(logger: Logger, id: string): Promise<string> {
+    if (this.verbose) {
+      await logger.logToStderr(`Getting the site URL of Microsoft 365 Group: ${id}...`);
+    }
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/groups/${id}/drive?$select=webUrl`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const res = await request.get<{ webUrl: string }>(requestOptions);
+    return res.webUrl.substring(0, res.webUrl.lastIndexOf('/'));
+  }
+
+  private async deleteM365GroupSite(logger: Logger, url: string): Promise<void> {
+    if (this.verbose) {
+      await logger.logToStderr(`Deleting the group site: '${url}'...`);
+    }
+
+    const spoAdminUrl = await spo.getSpoAdminUrl(logger, this.debug);
+    this.spoAdminUrl = spoAdminUrl;
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.spoAdminUrl}/_api/GroupSiteManager/Delete?siteUrl='${url}'`,
+      headers: {
+        'content-type': 'application/json;odata=nometadata',
+        accept: 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    await request.post(requestOptions);
+  }
+
+  private async deleteM365GroupFromRecyclebin(logger: Logger, id: string): Promise<void> {
+    const maxRetries = 10;
+    const intervalInMs: number = 6000;
+
+    for (let retries = 0; retries < maxRetries; retries++) {
+      if (await this.isM365GroupInDeletedItemsList(id)) {
```

Comment:
Do we need to check this on each call and resubmit the delete operation each time? Resubmitting deletion multiple times seems unnecessary. Shouldn't we submit it only once and then wait until the site disappears from deleting items (assuming that's the only way to check if it's actually been deleted)?

Reply:
The code does not resubmit the delete operation each time. It checks if the M365 group is in the deleted items list on each iteration, and if the group is found, it proceeds to remove it permanently and then exits the loop. There is no resubmission of the delete operation beyond this point.

I fear I may not have fully understood the comment. Could you please let me know how I can optimize it further?